### PR TITLE
[OSD-7882] Move getPodList functions out of drain strategies into pod…

### PR DESCRIPTION
**Pull request by T0MASD**
… package

### What type of PR is this?
_(refactor)_

### What this PR does / why we need it?

Apply DRY to PodList functions in drain strategies

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_ [OSD-7882](https://issues.redhat.com/browse/OSD-7882)

### Pre-checks (if applicable):
- [x] Ran `make test` command locally to validate code changes

### CODE REVIEW

File: pkg/drain/podDeleteStrategy.go [hunk 1]
```diff
@@ -16,7 +13,8 @@ type podDeletionStrategy struct {
 }
 
 func (pds *podDeletionStrategy) Execute(node *corev1.Node) (*DrainStrategyResult, error) {
-	podsToDelete, err := pds.getPodList(node)
+	filters := []pod.PodPredicate{isOnNode(node)}
```
Comment (mrbarge) [inline pkg/drain/podDeleteStrategy.go:16]: Any reason why no appending of the podDeletionStrategy filters here? ie.
```suggestion
	filters := append([]pod.PodPredicate{isOnNode(node)}, pds.filters...)
```
That seems to be inadvertently being dropped through this change. I notice it is being done for the `stuckTerminatingStrategy.go` file, but not here.

File: pkg/drain/podDeleteStrategy.go [hunk 2]
```diff
@@ -34,29 +32,11 @@ func (pds *podDeletionStrategy) Execute(node *corev1.Node) (*DrainStrategyResult
 }
 
 func (pds *podDeletionStrategy) IsValid(node *corev1.Node) (bool, error) {
-	targetPods, err := pds.getPodList(node)
+	filters := []pod.PodPredicate{isOnNode(node)}
```
Comment (mrbarge) [inline pkg/drain/podDeleteStrategy.go:35]: Same question/comment from earlier applies here.

File: pkg/drain/removeFinalizersStrategy.go [hunk 3]
```diff
@@ -16,7 +13,8 @@ type removeFinalizersStrategy struct {
 }
 
 func (rfs *removeFinalizersStrategy) Execute(node *corev1.Node) (*DrainStrategyResult, error) {
-	podsWithFinalizers, err := rfs.getPodList(node)
+	filters := []pod.PodPredicate{isOnNode(node), hasFinalizers}
```
Comment (mrbarge) [inline pkg/drain/removeFinalizersStrategy.go:16]: Same question/comment from earlier applies here.

File: pkg/drain/removeFinalizersStrategy.go [hunk 4]
```diff
@@ -33,29 +31,12 @@ func (rfs *removeFinalizersStrategy) Execute(node *corev1.Node) (*DrainStrategyR
 }
 
 func (rfs *removeFinalizersStrategy) IsValid(node *corev1.Node) (bool, error) {
-	targetPods, err := rfs.getPodList(node)
+	filters := []pod.PodPredicate{isOnNode(node), hasFinalizers}
```
Comment (mrbarge) [inline pkg/drain/removeFinalizersStrategy.go:34]: Same question/comment from earlier applies here.

File: pkg/pod/pod.go [hunk 5]
```diff
@@ -88,3 +89,22 @@ func RemoveFinalizersFromPod(c client.Client, pl *corev1.PodList) (*RemoveFinali
...
+
+	allPods := &corev1.PodList{}
+	err = c.List(context.TODO(), allPods, &client.ListOptions{
+		FieldSelector: fieldSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return FilterPods(allPods, filters...), nil
```
Comment (bmeng) [inline pkg/pod/pod.go:109]: it returns a function here? why cannot return a slice of pods instead?

Comment (T0MASD) [reply pkg/pod/pod.go:109]: Do you mean this part should be updated to:
```
filteredPods := FilterPods(allPods, filters...), nil
return filteredPods
```
FWIW the old function was also returning the function:
```
return pod.FilterPods(allPods, filters...), nil
```
Functions are returned in other places as seen from
```
$ egrep -Rn "return \w+\(.+\)" pkg/ | wc -l
31
```

Comment (mrbarge) [reply pkg/pod/pod.go:109]: It seems to be returning the output of that function (`*corev1.PodList`) not the function itself. 
Which seems fine to me given that the function itself only returns one value.

**Pull request merged**